tools/cp-utility/src/main.rs

```diff
@@ -42,19 +44,27 @@ struct CopyOperation {
 
 /// Parse command line arguments and transform into `CopyOperation`
 fn parse_args(args: Vec<&str>) -> io::Result<CopyOperation> {
-    if !(args.len() == 3 || args.len() == 4 && args[1].eq("-a")) {
+    if !(args.len() == 3 || (args.len() == 4 && (args[1] == "-a" || args[1] == "-r"))) {
         return Err(io::Error::new(
             io::ErrorKind::InvalidInput,
-            "Invalid parameters. Expected cp [-a] <source> <destination>",
+            "Invalid parameters. Expected cp [-a | -r] <source> <destination>",
         ));
     }
 
     if args.len() == 4 {
-        return Ok(CopyOperation {
-            source: PathBuf::from(args[2]),
-            destination: PathBuf::from(args[3]),
-            copy_type: CopyType::Archive,
-        });
+        if args[1].eq("-a") {
+            return Ok(CopyOperation {
+                source: PathBuf::from(args[2]),
+                destination: PathBuf::from(args[3]),
+                copy_type: CopyType::Archive,
+            });
+        } else if args[1].eq("-r") {
+            return Ok(CopyOperation {
+                source: PathBuf::from(args[2]),
+                destination: PathBuf::from(args[3]),
+                copy_type: CopyType::Recursive,
+            });
+        }
```

Comment:
Since these operations only differ by the `copy_type` field this can be simplified with a `match` statement.  (double check my syntax here)

```suggestion
           return Ok(CopyOperation {
              source: PathBuf::from(args[2]),
              destination: PathBuf::from(args[3]),
              copy_type: match args[1] {
                "-a" => CopyType::Archive,
                "-r" => CopyType::Recursive,
                _ => panic!("invalid argument")
                }
          });
```

Reply:
agreed, committed the changes